nautobot/extras/tasks.py

```diff
@@ -87,3 +92,78 @@ def provision_field(field_id, content_type_pk_set):
         for obj in model.objects.all():
             obj._custom_field_data[field.name] = field.default
             obj.save()
+
+
+@app.task()
+def process_webhook(webhook_pk, data, model_name, event, timestamp, username, request_id):
+    """
+    Make a POST request to the defined Webhook
+    """
+    from nautobot.extras.models import Webhook  # avoiding circular import
+
+    webhook = Webhook.objects.get(pk=webhook_pk)
+
+    context = {
+        "event": dict(ObjectChangeActionChoices)[event].lower(),
+        "timestamp": timestamp,
+        "model": model_name,
+        "username": username,
+        "request_id": request_id,
+        "data": data,
+    }
+
+    # Build the headers for the HTTP request
+    headers = {
+        "Content-Type": webhook.http_content_type,
+    }
+    try:
+        headers.update(webhook.render_headers(context))
+    except (TemplateError, ValueError) as e:
+        logger.error("Error parsing HTTP headers for webhook {}: {}".format(webhook, e))
+        raise e
+
+    # Render the request body
+    try:
+        body = webhook.render_body(context)
+    except TemplateError as e:
+        logger.error("Error rendering request body for webhook {}: {}".format(webhook, e))
```

Comment:
Pylint will eventually dislike this, as it prefers using %-formatting for log messages, i.e.:

```suggestion
        logger.error("Error rendering request body for webhook %s: %s", webhook, e)
```

Similar suggestion on the other logger calls in this function.